bindings/src/Capgemini.PowerApps.SpecFlowBindings/PowerAppsStepDefiner.cs

```diff
@@ -43,18 +43,26 @@ public abstract class PowerAppsStepDefiner
         private static IDictionary<string, string> userProfilesDirectories;
         private static object userProfilesDirectoriesLock = new object();
 
+        private static string accessToken;
+
+        /// <summary>
+        /// Date and Time when access token expires....
+        /// </summary>
+        private static DateTimeOffset accessTokenExpiresOn;
+
         /// <summary>
         /// Gets access token used to authenticate as the application user configured for testing.
         /// </summary>
         protected static string AccessToken
         {
             get
             {
-                var hostSegments = TestConfig.GetTestUrl().Host.Split('.');
+                if (accessToken == null)
+                {
+                    accessToken = GetAccessToken();
+                }
 
-                return GetApp().AcquireTokenForClient(new string[] { $"https://{hostSegments[0]}.api.{hostSegments[1]}.dynamics.com//.default" })
-                    .ExecuteAsync()
-                    .Result.AccessToken;
```

Comment:
This property is only ever going to return a single access token for the duration of a test run unless we nullify the `accessToken` field. It might be best to include the expiry logic in the getter here.

Reply:
But then it doesn't return a consistent value...

Reply:
Sorry, missed the assignment on line 178. So the field itself is getting updated, it's just the mechanism by which it get updated is not the most reliable. Anyone that's using the `AccessToken` directly for their own purposes (like we use it on our project to make requests to the Web API within bindings) won't be getting the benefit of token regeneration on expiry in this way. 

I'm almost done making some changes to address this.

Reply:
I think the issue before was that it wasn't possible to read it without also writing to it. But putting the expiry logic in the getter should be fine as that as a valid case for updating it.